src/main/webapp/stylesheets/teammatesCommon.css

```diff
@@ -1184,14 +1184,33 @@ Prevent Long URL From Breaking Out of Container
     margin-right: 5px;
 }
 
-/*
-Glyphicon for success status messages
-*/
+/*//////////////////////////////////////////////////////////
+////////////////////// Status Icons //////////////////////
+//////////////////////////////////////////////////////////*/
 
-.icon-success:before {
+.icon-success:before,
+.icon-warning:before,
+.icon-danger:before,
+.icon-info:before {
+    /*noinspection CssNoGenericFontName*/
```

Comment:
> This line by @whipermr5 pretty much sums up my stand. It is difficult to enforce even one IDE profile; you'd have noticed by now that the defaults of any IDE lack many good features of (at least Java) languages.

Well, I definitely do not think adding IDE specific comments is a good idea as well, but at the moment the most reasonable compromise I can think of is what I have discussed.


If there is an equivalent check that can be integrated into CI, we can very well turn that inspection off in IntelliJ (but it does not seem any popular tool does this).

>  I'm not saying that you should throw away good IDE features, but unless they are general enough (e.g other IDEs will also complain anyway; CheckStyle/... will complain anyway)

I think I have made arguments for that it is general enough unless is there some reason you don't think so?

> My own Eclipse inspection profile has many more additional rules on top of the default ones, and I'm not sure how many Eclipse violations are detected out there, but I got 245 right now (many of which I'm itching to fix).

Violations in Eclipse that are general enough should be reflected in IntelliJ as well, so we could always fix them (and if they aren't, we should turn those checks off). For example, using the diamond constructor for Java is definitely a good idea, but `CheckStyle` doesn't have a way to check for that.

That should be similar in this case.

> I don't think we have a ground to include any IDE-specific annotations around. It is like saying "I [a person] don't understand this code, let's add a comment here" while the majority of people out there have no problem understanding it.

I agree there certainly needs to be some kind of assumed knowledge so comments don't end up mentioning for example why inheritance is used but at least I think this is not the case here.

Do the majority of people understand that fonts should have generic fallbacks, and thus icon fonts are special cases? Assuming this knowledge does not seem ideal, but while that can be arguable, a linter would help us with avoid this kind of problem altogether.

So logically if we want to assume the user have that knowledge, we simply mention that in our style guide. And if we violate our style guide, we will logically need to mention why we choose to ignore our style guide (so it's like how people also serve the linting role when we don't follow our style guide).

But this causes the problem that the warnings still persist in IntelliJ then, should we then turn the inspection off and let people manually check, but that does not sound like a very good idea either.

So given that IntelliJ is the only one who gives such a warning right now, and that warning is general enough that's why I think the most reasonable compromise would be https://github.com/TEAMMATES/teammates/pull/7844#discussion_r130527179

--- 

Basically, I am not saying that this is a good solution, but it is the most reasonable compromise I can think of (otherwise there won't be a need for discussion anyway). I am open to a better alternative.

Ultimately, I know of a possible long term solution, but just not sure whether we can actually achieve it. We can use IntelliJ inspections as a supplement to the current static analysis tools we have, by integrating it into CI as well. Doing it through `TeamCity` would be reasonably, but the problem is that our project doesn't seem to qualify for the open source hosted version, and I'm not sure if self-hosting would be a good idea.

Installing IntelliJ and then running the inspections on Travis seems possible, but I'm not sure if we actually need the Ultimate version to have all the inspections.